src/vm/dllimport.cpp

```diff
@@ -3685,7 +3685,12 @@ static void CreateNDirectStubWorker(StubState*         pss,
         // The return structure secret arg comes first, however byvalue return is processed at
         // the end because it could be the HRESULT-swapped argument which always comes last.
         fMarshalReturnValueFirst = HasRetBuffArg(&msig);
-#endif
+
+#ifdef UNIX_X86_ABI
+        fMarshalReturnValueFirst |= HasRetBuffArgUnmanagedFixup(&msig);
```

Comment:
I would make this 

```
#ifdef UNIX_X86_ABI
    fMarshalReturnValueFirst = ... maybe even inline HasRetBuffArgUnmanagedFixup here since it is trivial and nothing else needs it ...
#else
    fMarshalReturnValueFirst = HasRetBuffArg(&msig); 
#endif
```